engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -112,11 +114,22 @@ object PartialWorkflowSources {
         case (None, None, None) => s"$WorkflowSourceKey or $WorkflowUrlKey needs to be supplied".invalidNel
       }
 
-      // workflow inputs
-      val workflowInputs: ErrorOr[Vector[WorkflowJson]] = getStringValue(WorkflowInputsKey) match {
+      case class InputsAndRequestedIds(workflowInputs: Vector[WorkflowJson], requestedIds: Option[Vector[WorkflowId]])
+
+      def unpackIdsString(idsString: Option[String]): ErrorOr[Option[Vector[WorkflowId]]] = {
```

Comment:
Would it be preferable to have the IDs be a JSON array and use an off the shelf parser?

Reply:
In terms of difficulty this ended up being more or less the same, but together with @salonishah11 's suggestion of using an empty vector to indicate unsupplied has allowed me to reuse the same workflow inputs code, so this worked out pretty nicely, IMO